README.md

```diff
@@ -32,610 +34,107 @@ For more information refer to [IntelliJ IDEA documentation](https://www.jetbrain
 
-* `-t` - profile threads separately. Each stack trace will end with a frame
-  that denotes a single thread.  
-  Example: `asprof -t 8983`
+For more details, please refer to
+[Profiling In Container Documentation](https://github.com/async-profiler/async-profiler/blob/master/docs/ConverterUsage.md).
 
-* `-s` - print simple class names instead of FQN.
+# Profiling Non-Java applications
 
-* `-n` - normalize names of hidden classes / lambdas.
+Unlike traditional Java profilers, async-profiler monitors non-Java threads (e.g., GC threads)
+and also shows native frames in Java stack traces. This enables it to work with C and C++
+applications.
 
-* `-g` - print method signatures.
+For more details, please refer to
+[Profiling Non-Java Applications](https://github.com/async-profiler/async-profiler/blob/master/docs/ProfilingNonJavaApplications.md).
 
-* `-a` - annotate JIT compiled methods with `_[j]`, inlined methods with `_[i]`, interpreted methods with `_[0]` and C1 compiled methods with `_[1]`.
+# JFR Visualization
 
-* `-l` - prepend library names to symbols, e.g. ``libjvm.so`JVM_DefineClassWithSource``.
+`jfr` output from async-profiler can be visualized in a human-readable format using multiple
+options, listed and explained in the
+[JFR Visualization Documentation](https://github.com/async-profiler/async-profiler/blob/master/docs/JfrVisualization.md).
 
-* `-o fmt` - specifies what information to dump when profiling ends.
-  `fmt` can be one of the following options:
-    - `traces[=N]` - dump call traces (at most N samples);
-    - `flat[=N]` - dump flat profile (top N hot methods);  
-      can be combined with `traces`, e.g. `traces=200,flat=200`
-    - `jfr` - dump events in Java Flight Recorder format readable by Java Mission Control.
-      This *does not* require JDK commercial features to be enabled.
-    - `collapsed` - dump collapsed call traces in the format used by
-      [FlameGraph](https://github.com/brendangregg/FlameGraph) script. This is
-      a collection of call stacks, where each line is a semicolon separated list
-      of frames followed by a counter.
-    - `flamegraph` - produce Flame Graph in HTML format.
-    - `tree` - produce Call Tree in HTML format.  
-      `--reverse` option will generate backtrace view.
+# Troubleshooting
 
-* `--total` - count the total value of the collected metric instead of the number of samples,
-  e.g. total allocation size.
+For known issues faced while running async-profiler and their detailed troubleshooting,
+please refer [here](https://github.com/async-profiler/async-profiler/blob/master/docs/Troubleshooting.md).
 
-* `--chunksize N`, `--chunktime N` - approximate size and time limits for a single JFR chunk.
-  A new chunk will be started whenever either limit is reached.
-  The default `chunksize` is 100MB, and the default `chunktime` is 1 hour.  
-  Example: `asprof -f profile.jfr --chunksize 100m --chunktime 1h 8983`
+# List of all async-profiler documentation
```

Comment:
This should be ordered and structured properly.
Creating a good structure for the documentation is one the main goals of this task.

Something like this:

1. Basic usage
  1.1. Getting Started
  1.2. Profiler Options
  1.3. Profiling Modes
  1.4. Other Use Cases <-- need a better name for it, may be Integrating Profiler?
  1.5. Profiling In Container

2. Profiler output
  2.1. Output Formats
  2.2. FlameGraph Interpretation
  2.3. JFR Visualization
  2.4. Converter

3. Advanced usage
  3.1. CPU Sampling Engines
  3.2. Stack Walking Modes
  3.3. Advanced Stacktrace Features

4. Profiling Non-Java Applications

5. Troubleshooting

Reply:
Addressed in [adc296d](https://github.com/async-profiler/async-profiler/pull/1029/commits/adc296d22ffe7bcf227f89b9b92e799a2f72f2f0)